src/rred_reports/masterfile.py

```diff
@@ -142,5 +150,14 @@ def join_masterfile_dfs(masterfile_dfs: dict[str, pd.DataFrame]) -> pd.DataFrame
     Returns:
         pd.DataFrame: joined masterfile
     """
-    pupil_teachers = pd.merge(masterfile_dfs["pupils"], masterfile_dfs["teachers"], on="rred_user_id")
-    return pd.merge(pupil_teachers, masterfile_dfs["schools"], on="school_id")
+    teacher_schools = pd.merge(masterfile_dfs["teachers"], masterfile_dfs["schools"], on="school_id")
+    return pd.merge(teacher_schools, masterfile_dfs["pupils"], on="rred_user_id")
+
+
+def masterfile_columns():
+    """List of all masterfile columns, in the expected order"""
+    pupil_no, user_id, *other_pupil_fields = Pupil.fields()
+    school_id, *other_school_fields = School.fields()
+    user_id, *other_teacher_fields, school_id = Teacher.fields()
```

Comment:
Might be worth prepending an underscore to one of the `school_id`s here just so it's slightly clearer what the variable contains? I realise it's probably the same value though

Reply:
good call, have made them different and asserted that they are the same as a sanity check